packages/console/src/pages/SignInExperience/components/Preview/index.tsx

```diff
@@ -29,10 +35,26 @@ function Preview({
   className,
 }: Props) {
   const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const { data: connectorGroups } = useConnectorGroups();
   const [language, setLanguage] = useState<LanguageTag>('en');
   const [mode, setMode] = useState<Theme>(Theme.Light);
   const [platform, setPlatform] = useState<PreviewPlatform>(PreviewPlatform.DesktopWeb);
   const { languages } = useUiLanguages();
+  const { data: connectorFactories } =
+    useSWR<ConnectorFactoryResponse[]>('api/connector-factories');
+
+  const isNativeTabEnabled = useMemo(() => {
+    const supportNativeTargets =
+      connectorFactories
+        ?.filter(({ platform }) => platform === ConnectorPlatform.Native)
+        .map(({ target }) => target) ?? [];
```

Comment:
platform === ConnectorPlatform.Native || platform === ConnectorPlatform.Web ?

Also feels like we could declare supportNativeTargets outside of the isNativeTabEnabled hook. As most likely connectorFactories won't change.

Reply:
updated with a new solution.